blog/features/replica-signed-queries.md

```diff
@@ -0,0 +1,115 @@
+
+Having a single replica signature in a query response doesn't completely solve the problem that query responses cannot be trusted, however, replica signed queries do help improve the security of queries since the recipient can validate that the response came from a replica on the correct subnet. The primary goal of this feature is to provide better guarantees against malicious boundary nodes, since replica signed queries make it impossible for a malicious boundary node to return false data. 
+
+## How do replica signed queries work?
+
+First, a list of node IDs and their corresponding public keys is put into the subnet's state tree so that they can be certified by the subnet. Users can obtain the certificate to validate the node's keys through an HTTP `read_state` call that includes a timestamp value. 
+
+To support replica signed queries, the query response format has been changed. The response to a query call adds a list with one signature for the returned response produced by the IC node that evaluated the query call.
+
+The actual portion of the call that is signed is the HttpQueryResponse from the execution of the call. If the call is replied to, the `reply` field is included. If the call is rejected, the `error_code`, `reject_code`, and `reject_message` fields will be included in the response. 
+
+If the query call resulted in a reply, the response is a CBOR (see [CBOR](https://internetcomputer.org/docs/current/references/ic-interface-spec/#cbor)) map with the following fields:
+
+- `status (text)`: "replied".
+  
+- `reply`: a CBOR map with the field arg (blob) which contains the reply data.
+  
+- `signatures ([+ node-signature])`: a list containing one node signature for the returned query response.
+
+If the call resulted in a reject, the response is a CBOR map with the following fields:
+
+- `status (text)`: "rejected".
+  
+- `reject_code (nat)`: The reject code (see Reject codes: https://internetcomputer.org/docs/current/references/ic-interface-spec/#reject-codes).
+  
+- `reject_message (text)`: a textual diagnostic message.
+  
+- `error_code (text)`: an optional implementation-specific textual error code (see Error codes: https://internetcomputer.org/docs/current/references/ic-interface-spec/#error-codes).
+  
+- `signatures ([+ node-signature])`: a list containing one node signature for the returned query response.
+
+The query's signature (whose type is denoted as node-signature) is a [CBOR](https://internetcomputer.org/docs/current/references/ic-interface-spec/#cbor) (see CBOR) map with the following fields:
+
+- `timestamp (nat)`: the timestamp of the signature.
+
+- `signature (blob)`: the actual signature.
+
+- `identity (principal)`: the principal of the node producing the signature.
+
+Queries also include a request ID, which is a SHA256 hash of the query's contents. With this `request_id` value, signatures for one query cannot be confused or exchanged for another.
```

Comment:
```suggestion
The actual parts that are signed include the query response from the execution layer, the timestamp and the request ID of the query, which is a SHA256 hash of the query's contents. With this `request_id` value, signatures for one query cannot be confused or exchanged for another.
```
I think we can explain what are signed altogether here after we show the `node-signature`.